action.yaml

```diff
@@ -14,21 +14,80 @@ inputs:
   description:
     description: "Additional information for the commit"
     required: false
-    default: "${{ github.event.head_commit.message }}"
+    default: "Artefacts CI check"
+  artefacts-url-api:
+    description: "Artefacts API URL"
+    required: false
+    default: "https://app.artefacts.com/api"
   working-dir:
-    description: "Working dir containing "
+    description: "Working dir containing"
     required: false
     default: "."
 
 runs:
   using: "composite"
   steps:
     - name: Install artefacts client
-      run: pip install artefacts-client --extra-index-url https://d5cw4z7oemmfd.cloudfront.net/pep503/
+      run: pip install artefacts-client --extra-index-url https://d5cw4z7oemmfd.cloudfront.net/pep503/ -q
+      shell: bash
+
+    - name: Set commit description
+      id: set-description
+      run: |
+        if [ -n "${{ github.event.head_commit.message }}" ]; then
+          description="${{ github.event.head_commit.message }}"
+        else
+          description="${{ inputs.description }}"
+        fi
+        echo "DESCRIPTION=$description" >> $GITHUB_ENV
       shell: bash
 
     - name: Run remote job
       id: run-job
       run: |
-        cd ${{inputs.working-dir}} && ARTEFACTS_KEY=${{ inputs.artefacts-api-key }} artefacts run-remote ${{ inputs.job-name }} --config ${{ inputs.config-file }} --description "${{ inputs.description }}"
+        cd ${{ inputs.working-dir }} && ARTEFACTS_KEY=${{ inputs.artefacts-api-key }} ARTEFACTS_API_URL=${{ inputs.artefacts-url-api }} artefacts run-remote ${{ inputs.job-name }} --config ${{ inputs.config-file }} --description "$DESCRIPTION"
+      shell: bash
+
+    - name: Wait for platform to register the job
+      run: sleep 60
+      shell: bash
+
+    - name: Extract project name from artefacts.yaml
+      id: extract-project-name
+      run: |
+        project_name=$(awk '/^project:/ {print $2}' ${{ inputs.config-file }})
```

Comment:
my awk is tired: does it assume anything on the position of a project name in the file? is it ok if it is not on the first line? Could there be an issue if the customer job name is called `project`

Reply:
let's try to make sure this wont will only match if there is no ` ` in front of `project` to avoid false positive on a job name or param name if the user doesn't write his yaml in the expected order

Reply:
alternatively we could use `jq` instead!

Reply:
"common sense" says don't regex a data format, after all

Reply:
indeed!